### DISCUSSION TITLE
jpg SOI corrupted

### DISCUSSION THREAD

**Discussion by steveoc** [spatie/laravel-medialibrary#2819] [Q&A] [answered]
using 8.10.2

Thank you for this awesome package.

If i upload a .jpg and then view it in browser (```return response()->file($media->getPath());```) or download (```return $media;```) it then I get an unreadable file. Looking at the downloaded file in a hex editor it looks like the image header footer are shifted by the corrupted file having an extra 1 or 2 bytes at start before SOI: FF D8 and no EOI: FF D9. Manually correcting this makes the jpg readable again.

Same behaviour on Windows development and Ubuntu production.

Any idea why this would be?
Steve

**Comment by freekmurze** [comment, marked as answer]
Not really, this package doesn't work on the files directly. Under the hood imagick / gd are being used to manipulate images.

**Reply by steveoc** [reply]
Thanks for reply. Pdfs uploaded seem ok to view/download. It was working fine previously not sure when this started happening. Don't quite know where else to look now?

**Reply by steveoc** [reply]
Calling ```ob_clean();``` before ```return $media;``` fixes the issue so i guess it's some fragment of buffer from elsewhere?
